### PR TITLE
fix: upgrade libs

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
This PR is needed to apply changes from ```https://github.com/eclipse-che/che-devfile-registry/pull/804```.

### CODE REVIEW

**Review by svor — APPROVED** [review]
LGTM

File: packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/index.ts [hunk 1]
```diff
@@ -366,7 +367,16 @@ export const actionCreators: ActionCreators = {
...
+        if (
+          common.helpers.errors.includesAxiosResponse(e) &&
+          isOAuthResponse(
+            (
+              e as {
+                response: AxiosResponse;
+              }
+            ).response.data,
+          )
+        ) {
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/index.ts:379]: This seems to me to be unnecessary changes

```suggestion
        if (common.helpers.errors.includesAxiosResponse(e) && isOAuthResponse(e.response.data)) {
```

File: packages/dashboard-backend/package.json [hunk 2]
```diff
@@ -26,8 +26,7 @@
   ],
   "license": "EPL-2.0",
   "dependencies": {
-    "@devfile/api": "^2.2.1-alpha-1667236163",
-    "@eclipse-che/che-devworkspace-generator": "7.78.0",
+    "@eclipse-che/che-devworkspace-generator": "next",
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-backend/package.json:29]: Ok

File: packages/dashboard-backend/src/devworkspaceClient/services/__tests__/devWorkspaceApi.spec.ts [hunk 3]
```diff
@@ -12,18 +12,18 @@
...
 import * as mockClient from '@kubernetes/client-node';
 import { CustomObjectsApi } from '@kubernetes/client-node';
 import { IncomingMessage } from 'http';
 
 import { DevWorkspaceApiService } from '@/devworkspaceClient/services/devWorkspaceApi';
+import {
+  devworkspaceGroup,
+  devworkspaceLatestVersion,
+  devworkspacePlural,
+} from '@/devworkspaceClient/services/helpers/devworkspace/constants';
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-backend/src/devworkspaceClient/services/__tests__/devWorkspaceApi.spec.ts:26]: @akurinnoy It doesn't work for me(build failed), so I provided separate constants. I will be glad if you provide a better solution in a separate commit.

File: packages/dashboard-backend/src/devworkspaceClient/services/helpers/devworkspace/constants.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+export const devworkspaceGroup = 'workspace.devfile.io';
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-backend/src/devworkspaceClient/services/helpers/devworkspace/constants.ts:13]: @akurinnoy It doesn't work for me(build failed), so I provided separate constants. I will be glad if you provide a better solution in a separate commit.

File: packages/dashboard-frontend/src/services/devfileApi/devfileApi.ts [hunk 5]
```diff
@@ -18,4 +18,46 @@ export { DevWorkspaceSpec } from './devWorkspace/spec';
...
+    format: string;
+  }>;
+  static getAttributeTypeMap(): {
+    name: string;
+    baseName: string;
+    type: string;
+    format: string;
+  }[];
+  constructor();
+}
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/services/devfileApi/devfileApi.ts:63]: @akurinnoy I couldn't import these declared classes from '@devfile/api', so I added them there.
 I will be glad if you provide a better solution as a separate commit.

**Comment by dmytro-ndp** [comment]
@olexii4 : hello,
smoke tests had failed on build of "**che-dashboard-image-amd64**" image: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/eclipse-che_che-dashboard/1021/pull-ci-eclipse-che-che-dashboard-main-v11-dashboard-happy-path/1734978937685020672
<details><summary>Build log</summary>


@eclipse-che/dashboard-frontend: export 'V1alpha2DevWorkspaceStatus' (reexported as 'DevWorkspaceStatus') was not found in '@devfile/api' (module has no exports)
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/index.ts 1:0-63 3:15-25
@eclipse-che/dashboard-frontend:  @ ./src/store/Workspaces/devWorkspaces/index.ts 9:0-55 143:21-35 636:16-30
@eclipse-che/dashboard-frontend:  @ ./src/store/index.ts
@eclipse-che/dashboard-frontend:  @ ./src/store/configureStore.ts 5:0-29 8:11-19
@eclipse-che/dashboard-frontend:  @ ./src/index.tsx 10:0-52 14:18-32
@eclipse-che/dashboard-frontend: ERROR in ../../node_modules/@devfile/api/dist/index.js 1:0-28
@eclipse-che/dashboard-frontend: Module not found: Error: Can't resolve './http/http' in '/dashboard/node_modules/@devfile/api/dist'
@eclipse-che/dashboard-frontend: Did you mean 'http.js'?
@eclipse-che/dashboard-frontend: BREAKING CHANGE: The request './http/http' failed to resolve only because it was resolved as fully specified
@eclipse-che/dashboard-frontend: (probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
@eclipse-che/dashboard-frontend: The extension in the request is mandatory for it to be fully specified.
@eclipse-che/dashboard-frontend: Add the extension to the request.
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/devfileApi.ts 1:0-80 1:0-80
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/index.ts 1:0-63 3:15-25
@eclipse-che/dashboard-frontend:  @ ./src/store/Workspaces/devWorkspaces/index.ts 9:0-55 143:21-35 636:16-30
@eclipse-che/dashboard-frontend:  @ ./src/store/index.ts
@eclipse-che/dashboard-frontend:  @ ./src/store/configureStore.ts 5:0-29 8:11-19
@eclipse-che/dashboard-frontend:  @ ./src/index.tsx 10:0-52 14:18-32
@eclipse-che/dashboard-frontend: ERROR in ../../node_modules/@devfile/api/dist/index.js 2:0-28
@eclipse-che/dashboard-frontend: Module not found: Error: Can't resolve './auth/auth' in '/dashboard/node_modules/@devfile/api/dist'
@eclipse-che/dashboard-frontend: Did you mean 'auth.js'?
@eclipse-che/dashboard-frontend: BREAKING CHANGE: The request './auth/auth' failed to resolve only because it was resolved as fully specified
@eclipse-che/dashboard-frontend: (probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
@eclipse-che/dashboard-frontend: The extension in the request is mandatory for it to be fully specified.
@eclipse-che/dashboard-frontend: Add the extension to the request.
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/devfileApi.ts 1:0-80 1:0-80
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/index.ts 1:0-63 3:15-25
@eclipse-che/dashboard-frontend:  @ ./src/store/Workspaces/devWorkspaces/index.ts 9:0-55 143:21-35 636:16-30
@eclipse-che/dashboard-frontend:  @ ./src/store/index.ts
@eclipse-che/dashboard-frontend:  @ ./src/store/configureStore.ts 5:0-29 8:11-19
@eclipse-che/dashboard-frontend:  @ ./src/index.tsx 10:0-52 14:18-32
@eclipse-che/dashboard-frontend: ERROR in ../../node_modules/@devfile/api/dist/index.js 3:0-29
@eclipse-che/dashboard-frontend: Module not found: Error: Can't resolve './models/all' in '/dashboard/node_modules/@devfile/api/dist'
@eclipse-che/dashboard-frontend: Did you mean 'all.js'?
@eclipse-che/dashboard-frontend: BREAKING CHANGE: The request './models/all' failed to resolve only because it was resolved as fully specified
@eclipse-che/dashboard-frontend: (probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
@eclipse-che/dashboard-frontend: The extension in the request is mandatory for it to be fully specified.
@eclipse-che/dashboard-frontend: Add the extension to the request.
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/devfileApi.ts 1:0-80 1:0-80
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/index.ts 1:0-63 3:15-25
@eclipse-che/dashboard-frontend:  @ ./src/store/Workspaces/devWorkspaces/index.ts 9:0-55 143:21-35 636:16-30
@eclipse-che/dashboard-frontend:  @ ./src/store/index.ts
@eclipse-che/dashboard-frontend:  @ ./src/store/configureStore.ts 5:0-29 8:11-19
@eclipse-che/dashboard-frontend:  @ ./src/index.tsx 10:0-52 14:18-32
@eclipse-che/dashboard-frontend: ERROR in ../../node_modules/@devfile/api/dist/index.js 4:0-54
@eclipse-che/dashboard-frontend: Module not found: Error: Can't resolve './configuration' in '/dashboard/node_modules/@devfile/api/dist'
@eclipse-che/dashboard-frontend: Did you mean 'configuration.js'?
@eclipse-che/dashboard-frontend: BREAKING CHANGE: The request './configuration' failed to resolve only because it was resolved as fully specified
@eclipse-che/dashboard-frontend: (probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
@eclipse-che/dashboard-frontend: The extension in the request is mandatory for it to be fully specified.
@eclipse-che/dashboard-frontend: Add the extension to the request.
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/devfileApi.ts 1:0-80 1:0-80
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/index.ts 1:0-63 3:15-25
@eclipse-che/dashboard-frontend:  @ ./src/store/Workspaces/devWorkspaces/index.ts 9:0-55 143:21-35 636:16-30
@eclipse-che/dashboard-frontend:  @ ./src/store/index.ts
@eclipse-che/dashboard-frontend:  @ ./src/store/configureStore.ts 5:0-29 8:11-19
@eclipse-che/dashboard-frontend:  @ ./src/index.tsx 10:0-52 14:18-32
@eclipse-che/dashboard-frontend: ERROR in ../../node_modules/@devfile/api/dist/index.js 5:0-33
@eclipse-che/dashboard-frontend: Module not found: Error: Can't resolve './apis/exception' in '/dashboard/node_modules/@devfile/api/dist'
@eclipse-che/dashboard-frontend: Did you mean 'exception.js'?
@eclipse-che/dashboard-frontend: BREAKING CHANGE: The request './apis/exception' failed to resolve only because it was resolved as fully specified
@eclipse-che/dashboard-frontend: (probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
@eclipse-che/dashboard-frontend: The extension in the request is mandatory for it to be fully specified.
@eclipse-che/dashboard-frontend: Add the extension to the request.
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/devfileApi.ts 1:0-80 1:0-80
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/index.ts 1:0-63 3:15-25
@eclipse-che/dashboard-frontend:  @ ./src/store/Workspaces/devWorkspaces/index.ts 9:0-55 143:21-35 636:16-30
@eclipse-che/dashboard-frontend:  @ ./src/store/index.ts
@eclipse-che/dashboard-frontend:  @ ./src/store/configureStore.ts 5:0-29 8:11-19
@eclipse-che/dashboard-frontend:  @ ./src/index.tsx 10:0-52 14:18-32
@eclipse-che/dashboard-frontend: ERROR in ../../node_modules/@devfile/api/dist/index.js 6:0-26
@eclipse-che/dashboard-frontend: Module not found: Error: Can't resolve './servers' in '/dashboard/node_modules/@devfile/api/dist'
@eclipse-che/dashboard-frontend: Did you mean 'servers.js'?
@eclipse-che/dashboard-frontend: BREAKING CHANGE: The request './servers' failed to resolve only because it was resolved as fully specified
@eclipse-che/dashboard-frontend: (probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
@eclipse-che/dashboard-frontend: The extension in the request is mandatory for it to be fully specified.
@eclipse-che/dashboard-frontend: Add the extension to the request.
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/devfileApi.ts 1:0-80 1:0-80
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/index.ts 1:0-63 3:15-25
@eclipse-che/dashboard-frontend:  @ ./src/store/Workspaces/devWorkspaces/index.ts 9:0-55 143:21-35 636:16-30
@eclipse-che/dashboard-frontend:  @ ./src/store/index.ts
@eclipse-che/dashboard-frontend:  @ ./src/store/configureStore.ts 5:0-29 8:11-19
@eclipse-che/dashboard-frontend:  @ ./src/index.tsx 10:0-52 14:18-32
@eclipse-che/dashboard-frontend: ERROR in ../../node_modules/@devfile/api/dist/index.js 7:0-47
@eclipse-che/dashboard-frontend: Module not found: Error: Can't resolve './apis/baseapi' in '/dashboard/node_modules/@devfile/api/dist'
@eclipse-che/dashboard-frontend: Did you mean 'baseapi.js'?
@eclipse-che/dashboard-frontend: BREAKING CHANGE: The request './apis/baseapi' failed to resolve only because it was resolved as fully specified
@eclipse-che/dashboard-frontend: (probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
@eclipse-che/dashboard-frontend: The extension in the request is mandatory for it to be fully specified.
@eclipse-che/dashboard-frontend: Add the extension to the request.
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/devfileApi.ts 1:0-80 1:0-80
@eclipse-che/dashboard-frontend:  @ ./src/services/devfileApi/index.ts 1:0-63 3:15-25
@eclipse-che/dashboard-frontend:  @ ./src/store/Workspaces/devWorkspaces/index.ts 9:0-55 143:21-35 636:16-30
@eclipse-che/dashboard-frontend:  @ ./src/store/index.ts
@eclipse-che/dashboard-frontend:  @ ./src/store/configureStore.ts 5:0-29 8:11-19
@eclipse-che/dashboard-frontend:  @ ./src/index.tsx 10:0-52 14:18-32
@eclipse-che/dashboard-frontend: 7 errors have detailed information that is not shown.
@eclipse-che/dashboard-frontend: Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.
@eclipse-che/dashboard-frontend: webpack 5.89.0 compiled with 7 errors and 1 warning in 45295 ms
@eclipse-che/dashboard-frontend: @eclipse-che/dashboard-frontend: error Command failed with exit code 1.
@eclipse-che/dashboard-frontend: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
 
   Lerna (powered by Nx)   Running target build for 2 projects failed
   Failed tasks:
   
   - @eclipse-che/dashboard-frontend:build
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error: build error: building at STEP "RUN yarn build": while running runtime: exit status 1
INFO[2023-12-13T16:56:06Z] Ran for 6m12s                                
ERRO[2023-12-13T16:56:06Z] Some steps failed:                           
ERRO[2023-12-13T16:56:06Z] 
  * could not run steps: step che-dashboard-image failed: error occurred handling build che-dashboard-image-amd64: the build che-dashboard-image-amd64 failed after 3m1s with reason DockerBuildFailed: Dockerfile build strategy has failed. 
INFO[2023-12-13T16:56:06Z] Reporting job state 'failed' with reason 'executing_graph:step_failed:building_project_image' 

</p>
</details> 

Could you please take a look?

**Review by akurinnoy — APPROVED** [review]
